DNN Platform/Library/Services/Log/EventLog/LogProperties.cs

```diff
@@ -2,17 +2,21 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information
 namespace DotNetNuke.Services.Log.EventLog
-{
-    using System;
-    using System.Collections;
-    using System.IO;
+{
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
     using System.Text;
-    using System.Xml;
+    using System.Xml;
 
+    using DotNetNuke.Abstractions.Logging;
     using DotNetNuke.Common.Utilities;
 
-    public class LogProperties : ArrayList
-    {
+    /// <inheritdoc />
+    public class LogProperties : ArrayList, ILogProperties
```

Comment:
As we moving forward, do we really need to create our own collection property?  Should this possible be marked obsolete to get around it?  Not sure of upstream usage

Reply:
If it wasn't for that `Summary` property, we could easily replace this with `IDictionary<string, string>` as @ahoefling suggested in a different chain.

One thing we need to keep in mind as we look toward future improvements is that adding members to interfaces is a breaking change, so we can't just create `AddLog(ILogProperties)` now and then deprecate it and add `AddLog(IDictionary<string, string>)` without it being a breaking change.

Reply:
I have a lot of thoughts about this, I really don't like the `LogProperties` and understand it was built during a time before C# generics. In the mobile/desktop world we typically have an event logger interface with the following API

```c#
void TrackEvent(string name, IDictionary<string, string> properties);
```

I would argue that `LogProperties` should be updated to meet that need but I don't think it is appropriate for this PR. My goal for this PR was a lift and shift for Dependency Injection. Once we accept this PR I think it will make sense to create a new API specification that simplifies EventLogging. I think the current API creates a barrier to entry as it requires a lot of guess and check work to create custom events. When using the API it should be as simple as the code snippet in this comment.

Is there any actionable item I need to take on this thread?